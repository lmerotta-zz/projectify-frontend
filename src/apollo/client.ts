import {
  ApolloClient,
  InMemoryCache,
  from,
  createHttpLink,
} from "@apollo/client";
import { RestLink } from "apollo-link-rest";
import qs from "qs";
import i18next from "i18next";

const restLink = new RestLink({
  endpoints: {
    login: {
      uri: `${process.env.REACT_APP_ACTIONS_URL}security/authentication/login`,
    },
    authorize: {
      uri: `${process.env.REACT_APP_ACTIONS_URL}security/authentication/authorize`,
      responseTransformer: (response: Response) => {
        const url = response.url;
        const decoded = qs.parse(url.slice(url.lastIndexOf("?") + 1));
        return {
          code: decoded.code,
        };
      },
    },
  },
  credentials: "include",
});

const httpLink = createHttpLink({
  uri: process.env.REACT_APP_GRAPHQL_URL,
});

let locale = "";

i18next.on("languageChanged", (lng) => {
  locale = lng;
});

const client = new ApolloClient({
  cache: new InMemoryCache(),
  headers: {
    "x-locale": locale,
  },
  link: from([restLink, httpLink]),
});

export default client;
