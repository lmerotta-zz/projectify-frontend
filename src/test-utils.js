import React from "react";
import { render } from "@testing-library/react";
import { CacheProvider } from "@emotion/react";
import createCache from "@emotion/cache";

const cache = createCache({ key: "iodajdoiwanoawdni" });
cache.compat = true;

const AllTheProviders = ({ children }) => {
  return <CacheProvider value={cache}>{children}</CacheProvider>;
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
