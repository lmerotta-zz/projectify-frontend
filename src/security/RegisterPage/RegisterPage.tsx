/** @jsxImportSource @emotion/react */
import "twin.macro";

import {
  Button,
  FormGroup,
  Input,
  Link,
  Form,
  FormRow,
  FormLabel,
} from "components";
import { FormProvider, useForm } from "react-hook-form";
import { yupResolver } from "@hookform/resolvers/yup";
import * as yup from "yup";
import { gql, useMutation } from "@apollo/client";
import { Trans, useTranslation } from "react-i18next";
import routePrefixes from "utils/routing-prefix";
import i18next from "i18next";
import mapViolationsToForm from "utils/mapViolationsToForm";
import {
  LeftPane,
  SectionSubTitle,
  SectionTitle,
  SubmitAdditionalLinkWrapper,
} from "security/components";
import { toast } from "react-toastify";
import { useHistory } from "react-router";
import {
  registerUserVariables,
  registerUser_createUser,
} from "apollo/types/registerUser";

const animationVariants = {
  initial: {
    opacity: 0,
  },
  in: {
    opacity: 1,
  },
  out: {
    opacity: 0,
  },
};

const schema = yup.object().shape({
  firstName: yup.string().required(),
  lastName: yup.string().required(),
  email: yup.string().email().required(),
  password: yup.string().required(),
  repeatPassword: yup
    .string()
    .required()
    .test({
      name: "equalTo",
      exclusive: false,
      message: i18next.t(
        "security.register_page.form.errors.password_mismatch"
      ),
      test: (value, context) => value === context.parent.password,
    }),
});

export const REGISTER_MUTATION = gql`
  mutation registerUser(
    $email: String!
    $firstName: String!
    $lastName: String!
    $password: String!
  ) {
    createUser(
      input: {
        email: $email
        firstName: $firstName
        lastName: $lastName
        password: $password
      }
    ) {
      clientMutationId
    }
  }
`;

type RegisterFormType = {
  email: string;
  firstName: string;
  lastName: string;
  password: string;
};

const RegisterPage = () => {
  const form = useForm<RegisterFormType>({
    resolver: yupResolver(schema),
  });

  const history = useHistory();

  const { t } = useTranslation();

  const [register] = useMutation<
    registerUser_createUser,
    registerUserVariables
  >(REGISTER_MUTATION, {
    onError: (e) => {
      /* istanbul ignore else */
      if (!mapViolationsToForm<RegisterFormType>(form.setError, e)) {
        toast.error(t("global.errors.internal-server-error"));
      }
    },
    onCompleted: () => {
      toast.success(t("security.register_page.message.user_created"));
      history.push(`${routePrefixes.security}/login`);
    },
  });

  return (
    <LeftPane
      variants={animationVariants}
      initial="initial"
      animate="in"
      exit="out"
    >
      <SectionTitle>{t("security.register_page.page_title")}</SectionTitle>
      <SectionSubTitle>
        {t("security.register_page.page_subtitle")}
      </SectionSubTitle>
      <FormProvider {...form}>
        <Form
          onSubmit={form.handleSubmit(async (data) => {
            await register({
              variables: data,
            });
          })}
        >
          <FormRow>
            <FormGroup>
              <FormLabel>
                {t("security.register_page.form.label_firstName")}
              </FormLabel>
              <Input name="firstName" type="text" ref={form.register} />
            </FormGroup>

            <FormGroup>
              <FormLabel>
                {t("security.register_page.form.label_lastName")}
              </FormLabel>
              <Input name="lastName" type="text" ref={form.register} />
            </FormGroup>
          </FormRow>
          <FormRow>
            <FormGroup>
              <FormLabel>
                {t("security.register_page.form.label_email")}
              </FormLabel>
              <Input name="email" type="email" ref={form.register} />
            </FormGroup>
          </FormRow>
          <FormRow>
            <FormGroup>
              <FormLabel>
                {t("security.register_page.form.label_password")}
              </FormLabel>
              <Input name="password" type="password" ref={form.register} />
            </FormGroup>
          </FormRow>
          <FormRow>
            <FormGroup>
              <FormLabel>
                {t("security.register_page.form.label_repeatPassword")}
              </FormLabel>
              <Input
                name="repeatPassword"
                type="password"
                ref={form.register}
              />
            </FormGroup>
          </FormRow>

          <FormRow>
            <FormGroup>
              <Button
                block
                data-testid="btn-register"
                type="submit"
                color="secondary"
                disabled={form.formState.isSubmitting}
              >
                {t("security.register_page.form.btn_register")}
              </Button>
            </FormGroup>
          </FormRow>
          <FormRow>
            <FormGroup>
              <SubmitAdditionalLinkWrapper>
                <Trans
                  i18nKey="security.register_page.form.existing_account_link"
                  components={{
                    Link: (
                      <Link
                        to={`${routePrefixes.security}/login`}
                        color="secondary"
                      />
                    ),
                  }}
                />
              </SubmitAdditionalLinkWrapper>
            </FormGroup>
          </FormRow>
        </Form>
      </FormProvider>
    </LeftPane>
  );
};

export default RegisterPage;
