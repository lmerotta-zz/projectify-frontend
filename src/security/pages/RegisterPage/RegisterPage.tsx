import { gql } from "@apollo/client";
import { yupResolver } from "@hookform/resolvers/yup";
import { Button, Grid, Link, Typography } from "@mui/material";
import { TextField } from "components";
import { useRegisterUserMutation } from "generated/graphql";
import i18next from "i18next";
import { FormProvider, useForm } from "react-hook-form";
import { Trans, useTranslation } from "react-i18next";
import { Link as RouterLink, useNavigate } from "react-router-dom";
import { toast } from "react-toastify";
import { RightPane, RightPaneFormWrapper } from "security/components";
import mapViolationsToForm from "utils/mapViolationsToForm";
import * as yup from "yup";

export const REGISTER_MUTATION = gql`
  mutation registerUser(
    $email: String!
    $firstName: String!
    $lastName: String!
    $password: String!
  ) {
    createUser(
      input: {
        email: $email
        firstName: $firstName
        lastName: $lastName
        password: $password
      }
    ) {
      clientMutationId
    }
  }
`;

type RegisterFormType = {
  email: string;
  firstName: string;
  lastName: string;
  password: string;
  repeatPassword: string;
};

const schema = yup.object().shape({
  firstName: yup.string().required(),
  lastName: yup.string().required(),
  email: yup.string().email().required(),
  password: yup.string().required(),
  repeatPassword: yup
    .string()
    .required()
    .test({
      name: "equalTo",
      exclusive: false,
      message: i18next.t(
        "security.register_page.form.errors.password_mismatch"
      ),
      test: (value, context) => value === context.parent.password,
    }),
});

const RegisterPage = () => {
  const form = useForm<RegisterFormType>({
    resolver: yupResolver(schema),
  });

  const navigate = useNavigate();
  const { t } = useTranslation();

  const [register] = useRegisterUserMutation({
    onError: (e) => {
      console.log(e);
      /* istanbul ignore else */
      if (!mapViolationsToForm(form.setError, e)) {
        toast.error(t("global.errors.internal-server-error"));
      }
    },
    onCompleted: () => {
      toast.success(t("security.register_page.message.user_created"));
      navigate("../login");
    },
  });

  return (
    <RightPane>
      <Typography variant="h4" component="h1" sx={{ fontWeight: "medium" }}>
        {t("security.register_page.page_title")}
      </Typography>
      <Typography
        variant="subtitle1"
        component="h3"
        color="neutral"
        sx={{ fontWeight: "medium" }}
      >
        {t("security.register_page.page_subtitle")}
      </Typography>
      <FormProvider {...form}>
        <RightPaneFormWrapper
          component="form"
          onSubmit={form.handleSubmit(
            async ({ repeatPassword: _, ...data }) => {
              await register({
                variables: data,
              });
            }
          )}
        >
          <Grid container spacing={3}>
            <Grid item xs={6}>
              <TextField
                id="firstName"
                label={t("security.register_page.form.label_firstName")}
                fullWidth
                variant="standard"
                {...form.register("firstName")}
              />
            </Grid>
            <Grid item xs={6}>
              <TextField
                id="lastName"
                label={t("security.register_page.form.label_lastName")}
                fullWidth
                variant="standard"
                {...form.register("lastName")}
              />
            </Grid>
            <Grid item xs={12}>
              <TextField
                type="email"
                id="email"
                label={t("security.register_page.form.label_email")}
                fullWidth
                variant="standard"
                {...form.register("email")}
              />
            </Grid>
            <Grid item xs={12}>
              <TextField
                id="password"
                type="password"
                label={t("security.register_page.form.label_password")}
                fullWidth
                variant="standard"
                {...form.register("password")}
              />
            </Grid>
            <Grid item xs={12}>
              <TextField
                id="repeatPassword"
                type="password"
                label={t("security.register_page.form.label_repeatPassword")}
                fullWidth
                variant="standard"
                {...form.register("repeatPassword")}
              />
            </Grid>
            <Grid item xs={12}>
              <Button
                type="submit"
                color="primary"
                variant="contained"
                fullWidth
                size="large"
                disabled={form.formState.isSubmitting}
              >
                {t("security.register_page.form.btn_register")}
              </Button>
              <Typography mt={2} color="neutral" variant="body2">
                <Trans
                  i18nKey="security.register_page.form.existing_account_link"
                  components={{
                    Link: (
                      <Link
                        component={RouterLink}
                        to="../login"
                        color="secondary"
                        underline="hover"
                        variant="inherit"
                      />
                    ),
                  }}
                />
              </Typography>
            </Grid>
          </Grid>
        </RightPaneFormWrapper>
      </FormProvider>
    </RightPane>
  );
};

export default RegisterPage;
