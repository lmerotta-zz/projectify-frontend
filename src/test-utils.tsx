import { MockedProvider, MockedProviderProps } from "@apollo/client/testing";
import {
  CircularProgress,
  StyledEngineProvider,
  ThemeProvider,
} from "@mui/material";
import { render as realRender, RenderOptions } from "@testing-library/react";
import { ReactElement, ReactNode, Suspense } from "react";
import { MemoryRouter, MemoryRouterProps } from "react-router";
import { Route, Routes } from "react-router-dom";
import { ToastContainer } from "react-toastify";
import theme from "theme";

function renderWithProviders(
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries"> & {
    routerProps?: MemoryRouterProps;
    graphqlProps?: MockedProviderProps;
  }
) {
  const Wrapper = ({ children }: { children?: ReactNode }) => {
    return (
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={theme}>
          <ToastContainer
            theme="colored"
            autoClose={3000}
            containerId="react-toastify"
          />
          <MockedProvider {...options?.graphqlProps} addTypename={false}>
            <MemoryRouter {...options?.routerProps}>
              <Suspense fallback={<CircularProgress />}>
                <Routes>
                  <Route
                    path="/*"
                    element={
                      <>
                        <h1>TESTSASATATAST</h1>
                        {children}
                      </>
                    }
                  />
                </Routes>
              </Suspense>
            </MemoryRouter>
          </MockedProvider>
        </ThemeProvider>
      </StyledEngineProvider>
    );
  };

  return realRender(ui, { wrapper: Wrapper, ...options });
}

export * from "@testing-library/react";
export { default as userEvent } from "@testing-library/user-event";
export { renderWithProviders };
