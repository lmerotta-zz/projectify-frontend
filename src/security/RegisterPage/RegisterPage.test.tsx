import { MockedProvider } from "@apollo/client/testing";
import { render, fireEvent, act } from "test-utils";
import { MemoryRouter, Router } from "react-router";
import { createMemoryHistory } from "history";
import RegisterPage, { REGISTER_MUTATION } from "./RegisterPage";
import routePrefixes from "utils/routing-prefix";
import { toast } from "react-toastify";

describe("RegisterPage unit tests", () => {
  it("Shows a global error on server error", async () => {
    const toastSpy = jest.spyOn(toast, "error");
    const mocks = [
      {
        request: {
          query: REGISTER_MUTATION,
          variables: {
            firstName: "test",
            lastName: "test",
            email: "test@test.com",
            password: "test",
            repeatPassword: "test",
          },
        },
        error: new Error("test"),
      },
    ];

    const result = render(
      <MockedProvider mocks={mocks as any} addTypename={false}>
        <MemoryRouter>
          <RegisterPage />
        </MemoryRouter>
      </MockedProvider>
    );

    expect(result.asFragment()).toMatchSnapshot();

    fireEvent.input(result.getByTestId("input-firstName"), {
      target: { value: "test" },
    });
    fireEvent.input(result.getByTestId("input-password"), {
      target: { value: "test" },
    });
    fireEvent.input(result.getByTestId("input-lastName"), {
      target: { value: "test" },
    });
    fireEvent.input(result.getByTestId("input-email"), {
      target: { value: "test@test.com" },
    });
    fireEvent.input(result.getByTestId("input-repeatPassword"), {
      target: { value: "test" },
    });

    await act(async () => {
      fireEvent.click(result.getByTestId("btn-register"));
    });

    await act(async () => {
      await new Promise((resolve) => setTimeout(resolve, 0));
    });

    expect(toastSpy).toHaveBeenCalled();
  });

  it("Redirects to the login page on success", async () => {
    const toastSpy = jest.spyOn(toast, "success");

    const mocks = [
      {
        request: {
          query: REGISTER_MUTATION,
          variables: {
            firstName: "test",
            lastName: "test",
            email: "test@test.com",
            password: "test",
          },
        },
        result: {},
      },
    ];

    const history = createMemoryHistory();
    history.push("/test");

    const result = render(
      <MockedProvider mocks={mocks as any} addTypename={false}>
        <Router history={history}>
          <RegisterPage />
        </Router>
      </MockedProvider>
    );

    fireEvent.input(result.getByTestId("input-firstName"), {
      target: { value: "test" },
    });
    fireEvent.input(result.getByTestId("input-password"), {
      target: { value: "test" },
    });
    fireEvent.input(result.getByTestId("input-lastName"), {
      target: { value: "test" },
    });
    fireEvent.input(result.getByTestId("input-email"), {
      target: { value: "test@test.com" },
    });
    fireEvent.input(result.getByTestId("input-repeatPassword"), {
      target: { value: "test" },
    });

    await act(async () => {
      fireEvent.click(result.getByTestId("btn-register"));
    });

    await act(async () => {
      await new Promise((resolve) => setTimeout(resolve, 0));
    });

    expect(toastSpy).toHaveBeenCalledWith(
      "security.register_page.message.user_created"
    );
    expect(history.location.pathname).toEqual(
      `${routePrefixes.security}/login`
    );
  });
});
