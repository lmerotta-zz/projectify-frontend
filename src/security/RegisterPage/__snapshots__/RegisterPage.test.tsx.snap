// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterPage unit tests Shows a global error on server error 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: 100%;
  -webkit-flex: 1 1 0%;
  -ms-flex: 1 1 0%;
  flex: 1 1 0%;
  --tw-bg-opacity: 1;
  background-color: rgba(255, 255, 255, var(--tw-bg-opacity));
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-left: 2rem;
  padding-right: 2rem;
  padding-top: 2.5rem;
  padding-bottom: 2.5rem;
}

@media (min-width: 768px) {
  .emotion-0 {
    padding-left: 12rem;
    padding-right: 12rem;
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    -webkit-justify-content: center;
    justify-content: center;
  }
}

@media (min-width: 1024px) {
  .emotion-0 {
    width: 33.333333%;
    padding-left: 4rem;
    padding-right: 4rem;
  }
}

.emotion-1 {
  font-size: 2.25rem;
  line-height: 2.5rem;
  --tw-text-opacity: 1;
  color: rgba(63, 63, 70, var(--tw-text-opacity));
  font-weight: 700;
}

@media (min-width: 768px) {
  .emotion-1 {
    font-size: 3rem;
    line-height: 1;
  }
}

.emotion-2 {
  font-size: 1.25rem;
  line-height: 1.75rem;
  --tw-text-opacity: 1;
  color: rgba(161, 161, 170, var(--tw-text-opacity));
  font-weight: 500;
  margin-bottom: 2.5rem;
}

@media (min-width: 768px) {
  .emotion-2 {
    font-size: 1.5rem;
    line-height: 2rem;
  }
}

@media (min-width: 1024px) {
  .emotion-2 {
    margin-bottom: 5rem;
  }
}

.emotion-3 {
  margin-bottom: 1.25rem;
}

@media (min-width: 1024px) {
  .emotion-3 {
    margin-bottom: 2rem;
  }
}

.emotion-4 {
  position: relative;
  outline: 2px solid transparent;
  outline-offset: 2px;
}

.emotion-5 {
  width: 100%;
  position: relative;
}

.emotion-6 {
  width: 100%;
  padding-top: 1rem;
  padding-bottom: 0.5rem;
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  border-width: 1px;
  border-radius: 0.25rem;
  --tw-border-opacity: 1;
  border-color: rgba(156, 163, 175, var(--tw-border-opacity));
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  outline: 2px solid transparent;
  outline-offset: 2px;
  transition-property: background-color,border-color,color,fill,stroke;
  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
  transition-duration: 150ms;
}

.emotion-6:focus {
  --tw-border-opacity: 1;
  border-color: rgba(147, 51, 234, var(--tw-border-opacity));
}

.emotion-6:focus+.label {
  --tw-text-opacity: 1;
  color: rgba(147, 51, 234, var(--tw-text-opacity));
  top: 0px;
  --tw-translate-y: 0.125rem;
  font-size: 0.75rem;
  line-height: 1rem;
}

.emotion-6:not(:placeholder-shown) {
  --tw-border-opacity: 1;
  border-color: rgba(147, 51, 234, var(--tw-border-opacity));
}

.emotion-6:not(:placeholder-shown)+.label {
  --tw-text-opacity: 1;
  color: rgba(147, 51, 234, var(--tw-text-opacity));
  top: 0px;
  --tw-translate-y: 0.125rem;
  font-size: 0.75rem;
  line-height: 1rem;
}

.emotion-7 {
  --tw-translate-x: 0;
  --tw-translate-y: -50%;
  --tw-rotate: 0;
  --tw-skew-x: 0;
  --tw-skew-y: 0;
  --tw-scale-x: 1;
  --tw-scale-y: 1;
  -webkit-transform: translateX(var(--tw-translate-x)) translateY(var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y));
  -moz-transform: translateX(var(--tw-translate-x)) translateY(var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y));
  -ms-transform: translateX(var(--tw-translate-x)) translateY(var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y));
  transform: translateX(var(--tw-translate-x)) translateY(var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y));
  position: absolute;
  top: 50%;
  left: 0.75rem;
  --tw-text-opacity: 1;
  color: rgba(161, 161, 170, var(--tw-text-opacity));
  cursor: text;
  transition-property: all;
  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
  transition-duration: 150ms;
  pointer-events: none;
}

.emotion-18 {
  margin-bottom: 2rem;
}

@media (min-width: 1024px) {
  .emotion-18 {
    margin-bottom: 2rem;
  }
}

.emotion-23 {
  margin-bottom: 2rem;
}

@media (min-width: 1024px) {
  .emotion-23 {
    margin-bottom: 0.5rem;
  }
}

.emotion-28 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  margin-bottom: 0.75rem;
}

.emotion-29 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -ms-flex-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  justify-content: space-evenly;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-30 {
  transition-property: background-color,border-color,color,fill,stroke,opacity,box-shadow,transform;
  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
  transition-duration: 150ms;
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  border-radius: 0.375rem;
  --tw-text-opacity: 1;
  color: rgba(255, 255, 255, var(--tw-text-opacity));
  --tw-bg-opacity: 1;
  background-color: rgba(147, 51, 234, var(--tw-bg-opacity));
  width: 100%;
  margin-bottom: 1.25rem;
  -webkit-flex: 1 1 0%;
  -ms-flex: 1 1 0%;
  flex: 1 1 0%;
  padding-top: 1rem;
  padding-bottom: 1rem;
  font-weight: 700;
}

.emotion-30:hover {
  --tw-bg-opacity: 1;
  background-color: rgba(126, 34, 206, var(--tw-bg-opacity));
}

.emotion-31 {
  font-size: 0.875rem;
  line-height: 1.25rem;
  --tw-text-opacity: 1;
  color: rgba(161, 161, 170, var(--tw-text-opacity));
}

<div
    class="emotion-0"
    style="opacity: 0;"
  >
    <h1
      class="emotion-1"
    >
      security.register_page.page_title
    </h1>
    <h2
      class="emotion-2"
    >
      security.register_page.page_subtitle
    </h2>
    <form>
      <div
        class="emotion-3"
      >
        <div
          class="emotion-4"
        >
          <div
            class="emotion-5"
          >
            <input
              class="emotion-6"
              data-testid="input-firstName"
              name="firstName"
              placeholder=" "
              type="text"
            />
            <label
              class="label emotion-7"
              data-testid="label-firstName"
            >
              security.register_page.form.label_firstName
            </label>
          </div>
        </div>
      </div>
      <div
        class="emotion-3"
      >
        <div
          class="emotion-4"
        >
          <div
            class="emotion-5"
          >
            <input
              class="emotion-6"
              data-testid="input-lastName"
              name="lastName"
              placeholder=" "
              type="text"
            />
            <label
              class="label emotion-7"
              data-testid="label-lastName"
            >
              security.register_page.form.label_lastName
            </label>
          </div>
        </div>
      </div>
      <div
        class="emotion-3"
      >
        <div
          class="emotion-4"
        >
          <div
            class="emotion-5"
          >
            <input
              class="emotion-6"
              data-testid="input-email"
              name="email"
              placeholder=" "
              type="email"
            />
            <label
              class="label emotion-7"
              data-testid="label-email"
            >
              security.register_page.form.label_email
            </label>
          </div>
        </div>
      </div>
      <div
        class="emotion-18"
      >
        <div
          class="emotion-4"
        >
          <div
            class="emotion-5"
          >
            <input
              class="emotion-6"
              data-testid="input-password"
              name="password"
              placeholder=" "
              type="password"
            />
            <label
              class="label emotion-7"
              data-testid="label-password"
            >
              security.register_page.form.label_password
            </label>
          </div>
        </div>
      </div>
      <div
        class="emotion-23"
      >
        <div
          class="emotion-4"
        >
          <div
            class="emotion-5"
          >
            <input
              class="emotion-6"
              data-testid="input-repeatPassword"
              name="repeatPassword"
              placeholder=" "
              type="password"
            />
            <label
              class="label emotion-7"
              data-testid="label-repeatPassword"
            >
              security.register_page.form.label_repeatPassword
            </label>
          </div>
        </div>
      </div>
      <div
        class="emotion-28"
      />
      <div
        class="emotion-29"
      >
        <button
          class="emotion-30"
          data-testid="btn-register"
          type="submit"
        >
          security.register_page.form.btn_register
        </button>
        <span
          class="emotion-31"
        >
          security.register_page.form.existing_account_link
        </span>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
